### ISSUE TITLE
compare_ssim() crop behaviour with gaussian weights

### ISSUE THREAD

**Issue by Kawue** [scikit-image/scikit-image#3652]
When using gaussian weights the crop method to calculate the mean still relies on the window size. So if you use both, a window size and gaussian weights the filter will have a size of 13 while the crop method uses the given window size.

I guess:
``` 
if gaussian_weights:
    win_size = 11
```
needs to be outside of: `if win_size is None:`

Also I don't get why it is set to 11 when a sigma of 1.5 will produce a window size of 13. If the filter is 13 shouldn't the crop also use 13?

**Comment by grlee77** [comment]
Hi @Kawue,

Thanks for pointing this out. sorry about taking so long to get back to you.

In the case of user-provided `sigma` for `gaussian_weights=True`, we should set `win_size` to match the Gaussian size that will be produced, rather than using a fixed value of 11 or leaving it at a user-specified value.

Internally, `ndimage.gaussian_filter` uses a filter radius, `r`:
```python
r = int(truncate*sigma + 0.5) # truncate defaults to 4 in scipy.ndimage
```
so we should set `win_size = 2*r + 1` in general for the Gaussian case. 


> If the filter is 13 shouldn't the crop also use 13?

I think it should, but the difference will be pretty small in practice for most images.  If we want both sigma = 1.5 **and** win_size = 11 to more exactly match what is stated in the paper, then we could modify the call to `gaussian_filter` to use `truncate=3.5` instead of the default of 4. 

In practice, for most images cropping one additional pixel at the boundary will not make a substantial difference in the mean SSIM. Here is one concrete example with a 512x512 image using gaussian_weights=True with crop for win_size of 13 vs. 11. 

```python
import numpy as np
from skimage import data, img_as_float
from skimage.util import crop
from skimage.measure import compare_ssim

rstate = np.random.RandomState(5)
x = img_as_float(data.camera())
xn = x + 0.05*rstate.standard_normal(x.shape)

mssim, ssim_full = compare_ssim(x, xn, full=True, gaussian_weights=True)

print(crop(ssim_full, 5).mean())  # crop using win_size 11
print(crop(ssim_full, 6).mean())  # crop using win_size 13
```
0.6853435539367254
0.685481083259942

**Comment by grlee77** [comment]
I made a PR with a fix in #3802.
